picky/src/x509/pkcs7/authenticode.rs

```diff
@@ -0,0 +1,1832 @@
+                || AuthenticodeError::NoCertificatesAssociatedWithIssuerAndSerialNumber {
+                    issuer: issuer_and_serial_number.issuer.clone(),
+                    serial_number: issuer_and_serial_number.serial_number.0 .0.clone(),
+                },
+            )
+    }
+
+    pub fn authenticode_verifier(&self) -> AuthenticodeValidator {
+        AuthenticodeValidator {
+            authenticode_signature: self,
+            inner: RefCell::new(AuthenticodeValidatorInner {
+                strictness: Default::default(),
+                now: None,
+                excluded_cert_authorities: vec![],
+                expected_file_hash: None,
+            }),
+        }
+    }
+
+    pub fn file_hash(&self) -> Option<Vec<u8>> {
+        let spc_indirect_data_content = match &self.0 .0.signed_data.content_info.content {
+            Some(content_value) => match &content_value.0 {
+                ContentValue::SpcIndirectDataContent(spc_indirect_data_content) => spc_indirect_data_content,
+                _ => return None,
+            },
+            None => return None,
+        };
+
+        Some(spc_indirect_data_content.message_digest.digest.0.clone())
+    }
+
+    pub fn authenticated_attributes(&self) -> &[Attribute] {
+        &self
+            .0
+            .signer_infos()
+            .first()
+            .expect("Exactly one SignerInfo should be present")
+            .signed_attrs
+            .0
+             .0
+    }
+
+    pub fn unauthenticated_attributes(&self) -> &[UnsignedAttribute] {
+        &self
+            .0
+             .0
+            .signed_data
+            .signers_infos
+            .0
+             .0
+            .first()
+            .expect("Exactly one SignerInfo should be present")
+            .unsigned_attrs
+            .0
+             .0
+    }
+}
+
+impl From<Pkcs7> for AuthenticodeSignature {
+    fn from(pkcs7: Pkcs7) -> Self {
+        AuthenticodeSignature(pkcs7)
+    }
+}
+
+impl From<AuthenticodeSignature> for Pkcs7 {
+    fn from(authenticode_signature: AuthenticodeSignature) -> Self {
+        authenticode_signature.0
+    }
+}
+
+#[derive(Debug, Clone)]
+struct AuthenticodeStrictness {
+    require_basic_authenticode_validation: bool,
+    require_signing_certificate_check: bool,
+    require_not_before_check: bool,
+    require_not_after_check: bool,
+    require_ca_verification_against_ctl: bool,
+    require_chain_check: bool,
+    exclude_specific_cert_authorities_from_ctl_check: bool,
+}
+
+impl Default for AuthenticodeStrictness {
+    fn default() -> Self {
+        AuthenticodeStrictness {
+            require_basic_authenticode_validation: false,
+            require_signing_certificate_check: false,
+            require_not_before_check: false,
+            require_not_after_check: false,
+            require_ca_verification_against_ctl: false,
+            require_chain_check: false,
+            exclude_specific_cert_authorities_from_ctl_check: false,
```

Comment:
I would prefer it safe by default and let user opt out if necessary. It’s generally a better security practice known as whitelisting.